### PR TITLE
secret: Test for max delta of service restart and secret rollout

### PR DESCRIPTION
**Pull request by zados**

### CODE REVIEW

File: src/check_secret.py [hunk 1]
```diff
@@ -120,14 +119,20 @@ def get_file_mtime(file_path: str) -> datetime:
...
-                           get_secret_file_time(args.timestamp_file_path)
-                           )
+
+    # Get the maximum delta of service restart time and secret file time
+    delta = max(
+        get_time_delta(service_restart_time), get_time_delta(
+            get_secret_file_time(
+                args.timestamp_file_path
+            )
+        )
```
Comment (seqizz) [inline src/check_secret.py:134]:
```suggestion
        get_time_delta(service_restart_time),
        get_time_delta(get_secret_file_time(args.timestamp_file_path))
```

File: src/check_secret.py [hunk 2]
```diff
@@ -120,14 +119,20 @@ def get_file_mtime(file_path: str) -> datetime:
...
         warning_name = args.service_name
     else:
-        check_delta = get_file_mtime(args.file_path)
+        service_restart_time = get_file_mtime(args.file_path)
         warning_name = 'Server'
-    delta = get_time_delta(check_delta,
-                           get_secret_file_time(args.timestamp_file_path)
-                           )
+
+    # Get the maximum delta of service restart time and secret file time
```
Comment (seqizz) [inline src/check_secret.py:128]:
```suggestion
    # Get the maximum delta of service restart time and secret file time
    # We need the oldest of these 2 values to compare with today
```

**Review by seqizz — APPROVED** [review]
2 optional suggestion, lgtm

File: src/check_secret.py [hunk 3]
```diff
@@ -120,26 +119,40 @@ def get_file_mtime(file_path: str) -> datetime:
...
-    elif delta.days > args.warning_days:
+    elif service_restart_delta.days > args.warning_days:
         print(
             f"WARNING - {warning_name} does not run with newest set of"
             " secrets"
         )
         exit(1)
+    elif secret_file_delta.days > args.critical_days:
+        print(
+            f"WARNING - New secrets were not deployed on this server for "
```
Comment (seqizz) [inline src/check_secret.py:146]: I think there is a mixup between warning and critical warnings here

Comment (zados) [reply src/check_secret.py]: Done!

**Pull request merged**